integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,201 @@
+import warnings
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+from haystack_integrations.utils.nvidia import NimBackend, url_validation
+
+_DEFAULT_MODEL = "nvidia/nv-rerankqa-mistral-4b-v3"
+
+_MODEL_ENDPOINT_MAP = {
+    "nvidia/nv-rerankqa-mistral-4b-v3": "https://ai.api.nvidia.com/v1/retrieval/nvidia/nv-rerankqa-mistral-4b-v3/reranking",
+}
+
+
+@component
+class NvidiaRanker:
+    """
+    A component for ranking documents using ranking models provided by
+    [NVIDIA NIMs](https://ai.nvidia.com).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.rankers.nvidia import NvidiaRanker
+
+    ranker = NvidiaRanker(
+        model="nvidia/nv-rerankqa-mistral-4b-v3",
+        api_key="your-api-key",
```

Comment:
This wrong, `api_key` must be a `Secret`. I would change this like so:

```suggestion
        api_key=Secret.from_token("your-api-key"),
```

You also need to add its import.